components/camel-ai/camel-langchain4j-web-search/src/main/java/org/apache/camel/component/langchain4j/web/search/LangChain4jWebSearchConfiguration.java

```diff
@@ -0,0 +1,199 @@
+import org.apache.camel.RuntimeCamelException;
+import org.apache.camel.spi.Configurer;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriParams;
+
+@Configurer
+@UriParams
+public class LangChain4jWebSearchConfiguration implements Cloneable {
+
+    @Metadata(required = true, autowired = true)
+    @UriParam
+    private WebSearchEngine webSearchEngine;
+
+    @Metadata(required = true, defaultValue = "CONTENT")
+    @UriParam
+    private LangChain4jWebSearchResultType resultType = LangChain4jWebSearchResultType.CONTENT;
+
+    @Metadata(required = true, defaultValue = "1")
+    @UriParam
+    private Integer maxResults = 1;
+    @UriParam
+    private String language;
+    @UriParam
+    private String geoLocation;
+    @UriParam
+    private Integer startPage;
+    @UriParam
+    private Integer startIndex;
+    @UriParam
+    private Boolean safeSearch;
+    @UriParam
+    @Metadata(autowired = true)
+    private Map<String, Object> additionalParams;
+
+    @UriParam(label = "advanced")
+    @Metadata(autowired = true)
+    private WebSearchRequest webSearchRequest;
+
+    public WebSearchEngine getWebSearchEngine() {
+        return webSearchEngine;
+    }
+
+    /**
+     * The {@link WebSearchEngine} engine to use. This is mandatory. Use one of the implementations from Langchain4j web
+     * search engines.
+     */
+    public void setWebSearchEngine(WebSearchEngine webSearchEngine) {
+        this.webSearchEngine = webSearchEngine;
+    }
+
+    public LangChain4jWebSearchResultType getResultType() {
+        return resultType;
+    }
+
+    /**
+     * The {@link #resultType} is the result type of the request. Valid values are
+     * LANGCHAIN4J_WEB_SEARCH_ORGANIC_RESULT, CONTENT, or SNIPPET. CONTENT is the default value; it will return a list
+     * of String . You can also specify to return either the Langchain4j Web Search Organic Result object (using
+     * `LANGCHAIN4J_WEB_SEARCH_ORGNAIC_RESULT`) or snippet (using `SNIPPET`) for each result. If {@link #maxResults} is
+     * equal to 1, the response will be a single object instead of a list.
+     */
+    public void setResultType(LangChain4jWebSearchResultType resultType) {
+        this.resultType = resultType;
+    }
+
+    /**
+     * The {@link #maxResults} is the expected number of results to be found if the search request were made. Each
+     * search engine may have a different limit for the maximum number of results that can be returned.
+     *
+     * @return
+     */
+    public Integer getMaxResults() {
+        return maxResults;
+    }
+
+    public void setMaxResults(Integer maxResults) {
+        this.maxResults = maxResults;
+    }
+
+    /**
+     * The {@link #language} is the desired language for search results is a string that indicates that the search
+     * client desires search results in the specified language. Each search engine may have a different set of supported
+     * languages.
```

Comment:
The sentence is quite long. I'm not sure how to split the sentence.

Would it be:

`The {@link #language} is the desired language for search results. It is a string that indicates that the search client desires search results in the specified language. Each search engine may have a different set of supported languages.`


Edit: in fact, several are based on the same structure rigth after and I do not understand either